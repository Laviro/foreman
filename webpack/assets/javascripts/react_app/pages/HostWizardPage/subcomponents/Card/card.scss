.foreman_host_wizard_card {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  transition: 0.25s;
  border-radius: 5px;
  height: 200px;
  min-width: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin-bottom: 40px;
  cursor: pointer;

  img {
    max-width: 60px;
    margin-bottom: 10px;
  }

  .title {
    font-weight: bold;
    font-size: 16px;
  }

  &.selected {
    border: 4px solid #1c85b7;
  }
}
