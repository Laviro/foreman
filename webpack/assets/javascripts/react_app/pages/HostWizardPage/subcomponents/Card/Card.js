import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { noop } from 'patternfly-react';
import './card.scss';

const Card = ({ logo, title, className, onClick }) => (
  <div
    tabIndex="-1"
    className={classNames('foreman_host_wizard_card', className)}
    onClick={onClick}
  >
    <img src={logo} alt="card logo" />
    <p className="title">{title}</p>
  </div>
);

Card.propTypes = {
  className: PropTypes.string,
  logo: PropTypes.string,
  onClick: PropTypes.func,
  title: PropTypes.string,
};

Card.defaultProps = {
  className: null,
  logo: '',
  onClick: noop,
  title: null,
};

export default Card;
