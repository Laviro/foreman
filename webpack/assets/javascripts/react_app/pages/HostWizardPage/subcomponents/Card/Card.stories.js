import React from 'react';
import { storiesOf } from '@storybook/react';
import Card from './Card';

storiesOf('Components/HostWizard/Card', module).add('Basic Card', () => (
  <div style={{ width: '200px', margin: '20px' }}>
    <Card
      title="Host Group"
      logo="https://image.flaticon.com/icons/png/128/1085/1085839.png"
    />
  </div>
));
