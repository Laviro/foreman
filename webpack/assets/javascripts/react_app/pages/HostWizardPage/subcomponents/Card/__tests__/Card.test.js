import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import Card from '../Card';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('Card', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(Card, fixtures));
});
