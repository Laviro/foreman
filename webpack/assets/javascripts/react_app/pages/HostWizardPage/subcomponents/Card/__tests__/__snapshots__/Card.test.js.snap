// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card rendering render without Props 1`] = `
<div
  className="foreman_host_wizard_card"
  onClick={[Function]}
  tabIndex="-1"
>
  <img
    alt="card logo"
    src=""
  />
  <p
    className="title"
  />
</div>
`;
